CHANGELOG.md

```diff
@@ -1,3 +1,11 @@
+# 0.24.0 (May 24, 2024)
+
+### Breaking Changes
+
+- Upgrade to opentelemetry 0.23. Refer to the upstream
+  [changelog](https://github.com/open-telemetry/opentelemetry-rust/releases/tag/opentelemetry-0.23.0)
+  for more information.
+
```

Comment:
The update from the 2018 edition to the 2021 edition is also considered a breaking change, although it doesn't actually break anything.

Reply:
How is it breaking? I'm genuinely asking, can't think of a way.

With rust's stability guarantees the only way I can see this could have an impact is if someone tried to build this with a pre-2021 compiler, but MSRV of this project is already higher.

Reply:
> With rust's stability guarantees the only way I can see this could have an impact is if someone tried to build this with a pre-2021 compiler, but MSRV of this project is already higher.

That's true. Please allow me extending my wording:

> The update from the 2018 edition to the 2021 edition is also considered a breaking change, although it doesn't actually break anything since we have specified the rust-version.

It's fine to no include this from breaking change list.

Reply:
I agree that the edition change has no bearing on downstream users, since the MSRV is already high enough.